src/arrayMethodLastIndexOf.js

```diff
@@ -5,8 +5,31 @@
  *
  */
 function applyCustomLastIndexOf() {
-  [].__proto__.lastIndexOf2 = function() {
-    // write code here
+  [].__proto__.lastIndexOf2 = function(searchElement, fromIndex = this.length) {
+    if (searchElement === undefined) {
+      return -1;
+    }
+    let index = fromIndex;
+    if (fromIndex < 0) {
+      if (fromIndex < (-this.length)) {
```

Comment:
Try to avoid nested `if` statements
As a solution, you could create one composed condition(above) to return -1. Then define `index` with one ternary.